common/djangoapps/enrollment/views.py

```diff
@@ -616,13 +616,14 @@ def post(self, request):
                 org = course_id.org
                 update_email_opt_in(request.user, org, email_opt_in)
 
+            log.info('The user %s has already been enrolled in course run %s.', username, course_id)
```

Comment:
@schenedx see our [previous attempt](https://github.com/edx/edx-platform/pull/13663/files#diff-a4a6f81c03f035ebb4441f342e27c62eL614) at resolving this. We've been through this exercise before.